rskj-core/src/main/java/co/rsk/core/RskAddress.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.core;
+
+import com.google.common.primitives.UnsignedBytes;
+import org.ethereum.rpc.TypeConverter;
+import org.spongycastle.util.encoders.Hex;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Immutable representation of an RSK address.
+ * It is a simple wrapper on the raw byte[].
+ *
+ * @author Ariel Mendelzon
+ */
+public final class RskAddress {
+
+    /**
+     * This is the size of an RSK address in bytes.
+     */
+    private static final int LENGTH_IN_BYTES = 20;
+
+    private static final RskAddress NULL_ADDRESS = new RskAddress(new byte[LENGTH_IN_BYTES]);
+
+    /**
+     * This compares using the lexicographical order of the sender unsigned bytes.
+     */
+    public static final Comparator<RskAddress> COMPARATOR = Comparator.comparing(
```

Comment:
Maybe this class could just implement `Comparable`

Reply:
We discussed this offline and didn't agree on this. My view is that addresses don't have a natural order (this implies implementing `Comparable`), but just an order for a specific feature (e.g. serializing always in the same order).
I think it's a minor issue, maybe we should leave this out for now.